src/snowflake/snowpark/modin/plugin/_internal/generator_utils.py

```diff
@@ -212,7 +223,12 @@ def generate_irregular_range(
         while start + periods * offset <= end:
             periods += 1
 
-    num_offsets = get_active_session().range(start=0, end=periods, step=1)
+    # TODO: SNOW-1646883 fix invalid identifier error when sql_simplifier_enabled is True
+    session = get_active_session()
+    sql_simplifier_enabled = session.sql_simplifier_enabled
+    session.sql_simplifier_enabled = False
+    num_offsets = session.range(start=0, end=periods, step=1)
+    session.sql_simplifier_enabled = sql_simplifier_enabled
```

Comment:
Can we add a comment for why we disable the sql_simplifier for using `session.range`?

Reply:
see line 226

Reply:
Oh missed that, resolving this comment